### ISSUE TITLE
LEDs with HM-MOD-UART

### ISSUE THREAD

**Issue by frank-f** [alexreinert/PCB#10]
Hi!

I'm trying to design a PCB based on your HB-RF-USB. I don't need compatibility with RPI-RF-MOD and I want to achieve a more compact form factor. What I would be interested in is the three LEDs of RPI-RF-MOD. As far as I understand, I could just add a few LEDs to my board, right? All I'd need to know is, if they're connected to VCC or Ground on the other side. Unfortunately in all RPI-RF-MOD pictures I could find the area is covered by a sticker, so I don't know where the traces end. Also looking at the FT232RL template was no help, since these lines are configured as I/O. Can you tell me how to connect the LEDs?
Also while looking over closed issues here, I found that the reset line should have been inverted - would it make sense to do that in my board or would that break compatibility with the driver you're using in piVCCU?

Thank you!

**Comment by frank-f** [comment]
I dug further and I think the FTDI chip is pulling the lines low to enable the LEDs. Can you confirm that?

I still don't know if I should adapt the inverted RESET line.

**Comment by alexreinert** [comment]
Your idea will not work as rhe LEDs are only driven if there is a RPI-RF-MOD is available.
Regarding the form factor: Maybe the HmIP-RFUSB (Non Telekom version) would be the right solution for you, it has now support for BidCos/HmIP dual mode, too.
If you proseed with your own pcb, please use a different USB ID, the USB ID in the template is registered to me and dedicated for my PCB.

**Comment by frank-f** [comment]
Too bad about the LEDs. I didn't know about HmIP-RFUSB yet. It does look interesting - if it can do HM and HMIP, the difference to my own PCB would just be an SMA connector. Maybe I'll pursue in that direction.
In case I'll have a PCB produced, would piVCCU even still recognize when using a different USB ID?

**Comment by alexreinert** [comment]
The HmIP-RFUSB was only able to communicate using HmIP until the last month. But eQ-3 released a new firmware, that supports both, HM (BidCos) and HmIP. Support for that is included in the current piVCCU testing release, which will hopefully be released as stable in the next days. Soldering a pigtail with SMA connector to it, is possible, too. (Of course you loose the original CE certification and guarantee in that case)
The different USB ID would not be recognized. But I can add the USB ID to the kernel modules, if you are respect the license of the HB-RF-USB and publish the PCB under CC BY-NC-SA 4.0 license.

**Comment by frank-f** [comment]
> Soldering a pigtail with SMA connector to it, is possible, too. (Of course you loose the original CE certification and guarantee in that case)

Well my current solution is a modified HM-MOD-UART as well. I guess, I don't have any warranty on that either. 😇
![IMG_8297 Medium](https://user-images.githubusercontent.com/9752843/146922099-84311828-7872-4e4a-9a19-7a27a8b72d50.jpeg)

> The different USB ID would not be recognized. But I can add the USB ID to the kernel modules, if you are respect the license of the HB-RF-USB and publish the PCB under CC BY-NC-SA 4.0 license.

That would be awesome. Could you also enable the LEDs for that USB ID or is that more complicated?
Of course, the source files will be published under the same license - wether I'll have it produced in the end or not.

**Comment by alexreinert** [comment]
Yes, this more complicated, as this is partly in the closed source binaries of eQ-3.

**Comment by frank-f** [comment]
Ah, too bad. I'll keep them in the board schematic, but leave them unpopulated then. Or maybe I'll use them as RXLED and TXLED or something.

**Comment by frank-f** [comment]
> The different USB ID would not be recognized. But I can add the USB ID to the kernel modules, if you are respect the license of the HB-RF-USB and publish the PCB under CC BY-NC-SA 4.0 license.

It took a little while longer because I was waiting on the USB PID, some parts and also I first started designing it into an 1551USB3 case, but then noticed that it should also fit into a smaller version.

I am not sure if all (or any?) of the mosfets are needed. I think, two were not needed for my project, so I left those out. I kept the one that seems to switch the HM-MOD-UART 3.3V line on and off. Does it work like that with your driver?

I named it [HM-1551USB2](https://github.com/frank-f/HM-1551USB2).

**Comment by frank-f** [comment]
@alexreinert I decided to order a small batch of my PCBs. Can you add my VID/PID to piVCCU or can easily I do that myself somehow? I'd like to test if my design working. Thank you in advance. :)
